MacroPepelelipa/MacroPepelelipa/Controller/Conflict Treatment/ConflictHandlerObject.swift

```diff
@@ -0,0 +1,155 @@
+        let baseController = UIApplication.shared.windows.first(where: { $0.isKeyWindow })?.rootViewController
+        
+        if let navigationController = baseController as? UINavigationController {
+            return navigationController.visibleViewController
+        }
+        
+        if let presented = baseController?.presentedViewController {
+            return presented
+        }
+        return baseController
+    }
+    
+    internal init() {
+        
+    }
+    
+    func chooseVersion(completionHandler: @escaping (DataVersion) -> Void) {
+        completionHandler(.cloud)
+    }
+    
+    func errDidOccur(err: Error) {
+        
+        if let error = err as? WorkspaceError {
+            workspaceErrorHandling(error: error)
+        }
+        
+        if let error = err as? NotebookError {
+            notebookErrorHandling(error: error)
+        }
+        
+        if let error = err as? NoteError {
+            noteErrorHandling(error: error)
+        }
+        
+        if let error = err as? PersistentError {
+            persistentContainerErrorHandling(error: error)
+        }
+        
+        if let error = err as? TextBoxError {
+            textBoxErrorHandling(error: error)
+        }
+        
+        if let error = err as? ImageBoxError {
+            imageBoxErrorHandling(error: error)
+        }
+        
+        if let error = err as? CKError {
+            cloudKitErrorHandling(error: error)
+        }
+    }
+    
+    private func workspaceErrorHandling(error: WorkspaceError) {
+        let title: String = "Error in Workspace".localized()
+
+        switch error {
+        case .failedToFetch:
+            presentAlertController(title: title, message: "Failed to Fetch".localized())
+        case .failedToParse:
+            presentAlertController(title: title, message: "Failed to Parse".localized())
+        case .workspaceWasNull:
+            presentAlertController(title: title, message: "Workspace was Null".localized())
+        }
+    }
+    
+    private func notebookErrorHandling(error: NotebookError) {
+        let title: String = "Error in Notebook".localized()
+        switch error {
+        case .failedToParse:
+            presentAlertController(title: title, message: "Failed to Parse".localized())
+        case .notebookWasNull:
+            presentAlertController(title: title, message: "Notebook was Null".localized())
+        }
+    }
+    
+    private func noteErrorHandling(error: NoteError) {
+        let title: String = "Error in Note".localized()
+        switch error {
+        case .failedToParse:
+            presentAlertController(title: title, message: "Failed to Parse".localized())
+        case .noteWasNull:
+            presentAlertController(title: title, message: "Note was Null".localized())
+        }
+    }
+    
+    private func persistentContainerErrorHandling(error: PersistentError) {
+        let title: String = "Error in Persistent Container".localized()
```

Comment:
Error is in persistent object, any savable object has an id and the id was null, not container